gateway/api_definition.go

```diff
@@ -301,6 +301,19 @@ type APIDefinitionLoader struct {
 // keyed to the Api version name, which is determined during routing to speed up lookups
 func (a APIDefinitionLoader) MakeSpec(def *nestedApiDefinition, logger *logrus.Entry) *APISpec {
 	spec := &APISpec{}
+	apiString, err := json.Marshal(def)
+	if err != nil {
+		logger.WithError(err).WithField("name", def.Name).Error("Failed to JSON marshal API definition")
+		return spec
+	}
+
+	sha256hash := sha256.Sum256(apiString)
+	// Unique API content ID, to check if we already have if it changed from previous sync
+	spec.Checksum = base64.URLEncoding.EncodeToString(sha256hash[:])
+
+	if a.Gw.checkSumExist(spec.Checksum) && a.Gw.isStarted() {
```

Comment:
As mentioned above, It feels that a lot of additional complexity with mutexes, new structures, and isStarted, can be replaced by:
```
for _, s := range gw.apisByID {
  if s.Checksum == spec.Checksum {
    return spec
  }
}
```